### PR TITLE
Rename report to qc-report and update qc report to 0.9.0

### PR DESCRIPTION
**Pull request by fbdtemme**
## Description

This PR removes references to "webreport" and renames those to qc report.
The report template is updated to 0.9.0. This includes:

- Rename "report" supertitle to "qc report"
- Add Reads per UMI figure in sequencing panel
- Replace "Median antibody molecules per cell" to "Average  antibody molecules per cell" for consistency
- Hide "data missing" placeholder messages for not yet implemented features.

Fixes: [EXE-1229](https://linear.app/pixelgen-technologies/issue/EXE-1229),[EXE-1230](https://linear.app/pixelgen-technologies/issue/EXE-1230),[EXE-1231](https://linear.app/pixelgen-technologies/issue/EXE-1231),[EXE-1232](https://linear.app/pixelgen-technologies/issue/EXE-1232)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

A new test case has been added for generating a frequency table of the reads per molecule.

- [x] tests/report/test_data_collection.py

## PR checklist:

- [ ] This comment contains a description of changes (with reason).
- [ ] My code follows the style guidelines of this project
- [ ] Make sure your code lints, is well-formatted and type checks
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If you've added a new stage - have you followed the pipeline CLI conventions in the [USAGE guide](../USAGE.md)
- [ ] [README.md](./README.md) is updated
- [ ] If a new tool or package is included, update poetry.lock, [the conda recipe](../conda-recipe/pixelator/meta.yaml) and [cited it properly](../CITATIONS.md)
- [ ] Include any new [authors/contributors](../AUTHORS.md)
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and documentation and corrected any misspellings
- [ ] Usage Documentation is updated
- [ ] If you are doing a [release](../RELEASING.md#Releasing), or a significant change to the code, update [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

**Review by johandahlberg — APPROVED** [review]
Looks good. Nice and clean changes.

File: src/pixelator/report/qcreport/builder.py [hunk 1]
```diff
@@ -82,21 +82,21 @@ def _load_template(self) -> Tuple[LxmlElement, LxmlElement]:
         body = document.find("body")
 
         if body is None:
-            raise AssertionError("Could not find body element in webreport template")
+            raise AssertionError("Could not find body element in QC report template")
```
Comment (ambarrio) [inline src/pixelator/report/qcreport/builder.py:85]: IS it easy to test for this so we cover it?

File: tests/report/test_report.py [hunk 2]
```diff
@@ -79,20 +79,24 @@ def figure_data(data_root) -> WebreportData:
     with open(data_dir / "antibody_counts.csv") as f:
         ab_counts = f.read()
 
-    webreport_data = WebreportData(
+    with open(data_dir / "reads-per-umi-frequency.csv") as f:
+        reads_per_umi_freq = f.read()
+
+    webreport_data = QCReportData(
```
Comment (ambarrio) [inline tests/report/test_report.py:85]: webreport_data -> qcreport_data ?

File: tests/report/test_report.py [hunk 3]
```diff
@@ -79,20 +79,24 @@ def figure_data(data_root) -> WebreportData:
...
         component_data=embedding_data,
         ranked_component_size=ranked_component_size_data,
         sequencing_saturation=seq_saturation,
         antibodies_per_cell=ab_per_cell,
         antibody_percentages=ab_pct,
         antibody_counts=ab_counts,
+        reads_per_umi_frequency=reads_per_umi_freq,
     )
     return webreport_data
```
Comment (ambarrio) [inline tests/report/test_report.py:94]: webreport_data -> qcreport_data ?

File: src/pixelator/test_utils/report.py [hunk 4]
```diff
@@ -59,6 +60,7 @@ def test_report_run(self):
 
     @pytest.mark.dependency(depends=["test_report_run"])
     def test_report_customer_exists(self):
-        report_files = (self.workdir / "report").glob("*_report.html")
+        """Check if the qc report html exists."""
+        report_files = (self.workdir / "report").glob("*.qc-report.html")
```
Comment (ambarrio) [inline src/pixelator/test_utils/report.py:64]: I don't understand how this cannot be tested if L65 and L66 are tested. Are we not testing the function at all?

**Review by ambarrio — APPROVED** [review]
Some comments and questions with approval so you can move forward afterwards.

File: src/pixelator/report/qcreport/builder.py (hunk 1 above)
Comment (fbdtemme) [reply src/pixelator/report/qcreport/builder.py:85]: This assertion error should basically never happen. That would be an issue with the qc-report html template file.
This not happening is tested with eg: test_write_webreport

File: src/pixelator/cli/preqc.py [hunk 5]
```diff
@@ -130,10 +129,9 @@ def preqc(
     output,
     design,
 ):
-    """
-    Process raw Molecular Pixelation data (FASTQ) to perform QC, filtering,
-    trimming and removal of duplicates
-    """
+    """Process raw Molecular Pixelation data (FASTQ) to perform QC, filtering,
+    trimming and removal of duplicates.
+    """  # noqa: D205,D400,D415
```
[the remark this answers is not in the document]
Comment (johandahlberg) [reply src/pixelator/cli/preqc.py:134]: I actually added a general exception from these rules in my coming PR, for things under `pixelator/cli/` to try to not have to deal with this issue on all the click help texts.

[the remark this answers is not in the document]
Comment (fbdtemme) [reply src/pixelator/cli/preqc.py:134]: Ah nice.

File: tests/report/test_report.py (hunk 3 above)
Comment (fbdtemme) [reply tests/report/test_report.py:94]: fixed

File: src/pixelator/test_utils/report.py (hunk 4 above)
Comment (fbdtemme) [reply src/pixelator/test_utils/report.py]: Good catch, this test is indeed useless as is. I have updated it.

**Comment by fbdtemme** [comment]
I will wait with merging this untill tomorrow to make sure the report template is the exact same version as a the release in pixelator-qc-report.

**Pull request merged**